src/java/org/apache/cassandra/cql3/constraints/ColumnConstraintScalar.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.constraints;
+
+
+import java.io.IOException;
+
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.cql3.CqlBuilder;
+import org.apache.cassandra.cql3.Operator;
+import org.apache.cassandra.db.TypeSizes;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.io.IVersionedSerializer;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.schema.ColumnMetadata;
+
+public class ColumnConstraintScalar implements ColumnConstraint
+{
+    public final ColumnIdentifier param;
+    public final Operator relationType;
+    public final String term;
+
+    public final static Serializer serializer = new Serializer();
+
+    public final static class Raw
+    {
+        public final ColumnIdentifier param;
+        public final Operator relationType;
+        public final String term;
+
+        public Raw(ColumnIdentifier param, Operator relationType, String term)
+        {
+            this.param = param;
+            this.relationType = relationType;
+            this.term = term;
+        }
+
+        public ColumnConstraintScalar prepare()
+        {
+            return new ColumnConstraintScalar(param, relationType, term);
+        }
+    }
+
+    public ColumnConstraintScalar(ColumnIdentifier param, Operator relationType, String term)
+    {
+        this.param = param;
+        this.relationType = relationType;
+        this.term = term;
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Object columnValue)
+    {
+        Number columnValueNumber;
+        float sizeConstraint;
+
+        try
+        {
+            columnValueNumber = (Number) columnValue;
```

Comment:
Bold move. Not sure about the specifics but I would call `validate` at the beginning of this method. If somebody does not call `validate` before calling `evaluate` then if `columnValue` is not a number then we have a problem.

Reply:
As I see, you are calling `validate` way before `evaluate` ... yeah, right. It is not applicable to call `validate` so late / upon actual evaluation.